packages/kbn-grid-layout/grid/grid_height_smoother.tsx

```diff
@@ -22,8 +26,25 @@ export const GridHeightSmoother = ({
     const subscription = combineLatest([
       gridLayoutStateManager.gridDimensions$,
       gridLayoutStateManager.interactionEvent$,
-    ]).subscribe(([dimensions, interactionEvent]) => {
+      gridLayoutStateManager.expandedPanelId$,
+    ]).subscribe(([dimensions, interactionEvent, expandedPanelId]) => {
       if (!smoothHeightRef.current) return;
+
+      if (expandedPanelId) {
+        const viewPortHeight = getViewportHeight();
+        const smoothHeightRefY = smoothHeightRef.current.getBoundingClientRect().y;
+
+        // When panel is expanded, ensure the page occupies the full viewport height, no more, no less, so
+        // smoothHeight height = viewport height - smoothHeight position - EuiPanel padding.
+
+        const height = viewPortHeight - smoothHeightRefY - parseFloat(euiThemeVars.euiSizeL);
+        smoothHeightRef.current.style.height = height + 'px';
+        smoothHeightRef.current.style.transition = 'none';
+        return;
+      } else {
+        smoothHeightRef.current.style.transition = '';
```

Comment:
`transition: 'none'` explicitly disables animations, while `transition: ''` removes the inline transition style, allowing the default CSS-defined transition to apply.

In this component, the default transition is defined on line 72:
`transition: height 500ms linear;`

For my use case (expanding a panel), the default transition caused an odd animation effect (see the gif). To fix this, I disabled the transition with `smoothHeightRef.current.style.transition = 'none'`. Once the panel is minimized, I reset the inline style with transition: '', so the default transition is applied again.

I couldn't observe when this default transition takes place but I guess it is somewhere and first and foremost I don't want to break anything existing :D 

![Nov-22-2024 10-47-38](https://github.com/user-attachments/assets/4e1f5076-c803-4ef4-93a2-6f783990b4cc)

Reply:
Ah, gotcha!! 

> I couldn't observe when this default transition takes place but I guess it is somewhere and first and foremost I don't want to break anything existing :D

`transition: height 500ms linear` is so that, when when shrinking a panel causes the height of the viewport to shrink, we get a nice transition on drop:


https://github.com/user-attachments/assets/b33b7cb2-ccb2-4cd4-ba0c-bc9b18585e79



Good catch :) However, testing this on your branch, it seems we've lost this effect now? Can you look into that? 🙇 

https://github.com/user-attachments/assets/6690017b-2685-41a5-bcef-2a919c84148f

Reply:
Nice catch! 600e1779b55b9921f4dc624358c5a830a044718c fixed here